src/components/RecipientListItem.vue

```diff
@@ -29,6 +30,16 @@ export default {
 			required: true,
 		},
 	},
+	data() {
+		return {
+			isInternal: true,
+		}
+	},
+	async onMounted() {
+		if (this.$store.getters.getPreference('internal-addresses', 'false') === 'true') {
+			this.isInternal = await isInternal(this.option.email)
```

Comment:
I think we should push this to the server, e.g. to `\OCA\Mail\IMAP\PreviewEnhancer::process`. Otherwise there will be one request per recipient. That does not scale

Reply:
Alternative: process on the front-end

The list of internal addresses is likely very short with 1-5 domains. Load the list with initial state and do the checks in the front-end.

Reply:
I'm utilising the store instead now